lib/ably/realtime/channel.rb

```diff
@@ -179,15 +179,28 @@ def presence
         @presence
       end
 
-      # Return the message history of the channel
+      # Return the message history of the channel.
+      #
+      # Once attached to a channel, you can retrieve messages published on the channel before the
+      # channel was attached with the option <tt>:end => :before_attach</tt>.  This is very useful for
+      # developers who wish to display both new real-time messages as well as historical messages with
+      # the guarantee that no messages have been missed.
+      #
       #
       # @param (see Ably::Rest::Channel#history)
       # @option options (see Ably::Rest::Channel#history)
+      # @option options [Integer,Time,Symbol]   :end        Time or millisecond since epoch, or +:before_attach+ for all messages published before this channel has been attached.
       #
       # @yield [Ably::Models::PaginatedResource<Ably::Models::Message>] An Array of {Ably::Models::Message} objects that supports paging (#next_page, #first_page)
       #
       # @return [Ably::Util::SafeDeferrable]
       def history(options = {}, &callback)
+        if options[:end] == :before_attach
+          raise ArgumentError, 'option :before_attach cannot be specified if the channel is not attached' unless attached?
```

Comment:
> An exception isn't an inconvenience to a programmer; it's a way to tell him that the program he wrote cannot meaningfully continue. So if an exception is called for, it helps him.

Raising an exception here is an inconvenience because the program _can_ continue. `before_attach` just means that we want to retrieve any message before the attach event occurred. In our case, we can very well do that since we are not attached yet.